protocol/communities_messenger_token_permissions_test.go

```diff
@@ -936,3 +936,261 @@ func (s *MessengerCommunitiesTokenPermissionsSuite) TestViewChannelPermissions()
 	s.Require().Len(response.Messages(), 1)
 	s.Require().Equal(msg.Text, response.Messages()[0].Text)
 }
+
+func (s *MessengerCommunitiesTokenPermissionsSuite) testReevaluateMemberPrivilegedRoleInOpenCommunity(permissionType protobuf.CommunityTokenPermission_Type) {
+	community, _ := s.createCommunity()
+
+	createTokenPermission := &requests.CreateCommunityTokenPermission{
+		CommunityID: community.ID(),
+		Type:        permissionType,
+		TokenCriteria: []*protobuf.TokenCriteria{
+			&protobuf.TokenCriteria{
+				Type:              protobuf.CommunityTokenType_ERC20,
+				ContractAddresses: map[uint64]string{testChainID1: "0x123"},
+				Symbol:            "TEST",
+				Amount:            "100",
+				Decimals:          uint64(18),
+			},
+		},
+	}
+
+	response, err := s.owner.CreateCommunityTokenPermission(createTokenPermission)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+	s.Require().Len(response.Communities(), 1)
+	s.Require().True(response.Communities()[0].HasTokenPermissions())
+
+	waitOnCommunityPermissionCreated := s.waitOnCommunitiesEvent(s.owner, func(sub *communities.Subscription) bool {
+		return sub.Community.HasTokenPermissions()
+	})
+
+	err = <-waitOnCommunityPermissionCreated
+	s.Require().NoError(err)
+
+	community, err = s.owner.communitiesManager.GetByID(community.ID())
+	s.Require().NoError(err)
+	s.Require().True(community.HasTokenPermissions())
+
+	s.advertiseCommunityTo(community, s.alice)
+
+	var tokenPermission *protobuf.CommunityTokenPermission
+	for _, tokenPermission = range community.TokenPermissions() {
+		break
+	}
+
+	s.makeAddressSatisfyTheCriteria(testChainID1, aliceAddress1, tokenPermission.TokenCriteria[0])
+
+	s.joinCommunity(community, s.alice, alicePassword, []string{aliceAddress1})
+
+	community, err = s.owner.communitiesManager.GetByID(community.ID())
+	s.Require().NoError(err)
+	switch permissionType {
+	case protobuf.CommunityTokenPermission_BECOME_ADMIN:
+		s.Require().True(community.IsMemberAdmin(&s.alice.identity.PublicKey))
+	case protobuf.CommunityTokenPermission_BECOME_TOKEN_MASTER:
+		s.Require().True(community.IsMemberTokenMaster(&s.alice.identity.PublicKey))
+	default:
+		s.Require().True(false, "Unknown permission, please, update the test")
+	}
+
+	err = s.owner.communitiesManager.ReevaluateMembers(community)
+	s.Require().NoError(err)
```

Comment:
Why this is done here?

Reply:
In order to validate the scenario, when the user joined the community and after some time control node will call `ReevaluateMembers`. Here we are checking, that `ReevaluateMembers` fn works correctly when the user fits the role and his role is not changed.

Reply:
> fn works correctly when the user fits the role and **his role is not changed**.

My point was that only `err` is checked here. Shouldn't we also check that roles have not changed?